src/util/replace_symbol.cpp

```diff
@@ -29,7 +29,8 @@ void replace_symbolt::insert(
 
 bool replace_symbolt::replace(
   exprt &dest,
-  const bool replace_with_const) const
+  const bool replace_with_const,
+  const bool ignore_type_change) const
 {
```

Comment:
I think this is getting to bizarre, and to linker specific. I'd make that a separate piece of code, just for the linker.

Reply:
Work in progress, #2720 and #2721 are the first PRs to enable this. More to come.